### PR TITLE
Check that arch is supported when enabling livepatch

### LINKED ISSUE #93: Livepatch is only available for x86_64/amd64
Since livepatch is only available for x86_64/amd64 (64 bits, therefore), the ua-tools script could detect that and prevent the user from trying to enable it on an unsupported architecture.

For example, on s390x it's listed as available:
```
ubuntu@xenial-andreas:~$ ua status
esm: disabled (not available)
livepatch: disabled
fips: disabled
```

But won't install:
```
ubuntu@xenial-andreas:~$ sudo ua enable-livepatch <redacted>
Installing the canonical-livepatch snap.
This may take a few minutes depending on your bandwidth.
error: snap "canonical-livepatch" not found
```

uname:
```
ubuntu@xenial-andreas:~$ uname -a
Linux xenial-andreas 4.4.0-103-generic #126-Ubuntu SMP Mon Dec 4 16:22:13 UTC 2017 s390x s390x s390x GNU/Linux
```

### PR DESCRIPTION
**Pull request by albertodonato**
Fixes #93

### CODE REVIEW

File: modules/service-esm.sh [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # shellcheck disable=SC2039
 
 ESM_SUPPORTED_SERIES="precise"
+ESM_SUPPORTED_ARCHS=""
```
Comment (panlinux) [inline modules/service-esm.sh]: I think using a special key here is better than an empty string. How about "ALL", or "@all@", or something like that?

File: tests/test_livepatch.py [hunk 2]
```diff
@@ -42,6 +43,15 @@ def test_enable_livepatch_invalid_token(self):
         self.assertEqual(3, process.returncode)
         self.assertIn('Invalid or missing Livepatch token', process.stderr)
 
+    def test_enable_livepatch_unsupported_on_non_x86_64(self):
+        """Livepatch is only supported on x86_64."""
```
Comment (panlinux) [inline tests/test_livepatch.py]: A more appropriate test name and docstring would be that livepatch is not supported on i686.

File: modules/service-esm.sh (hunk 1 above)
Comment (panlinux) [inline modules/service-esm.sh]: Also, I don't think the case of an empty list is tested. You could write a test that sets the arch to something exotic and that esm is still enabled (or using the new ALL special name, if you decide to accept that suggestion).

Comment (albertodonato) [reply modules/service-esm.sh]: "ALL" sounds good to me, as it matches deb archs. Done

File: tests/test_livepatch.py (hunk 2 above)
Comment (albertodonato) [reply tests/test_livepatch.py]: fixed

File: modules/service-esm.sh (hunk 1 above)
Comment (albertodonato) [reply modules/service-esm.sh]: added a test

**Pull request merged**